### PR TITLE
feat(yt-shorts): add youtube shorts downloader service

### PR DESCRIPTION
**Pull request by arpanetus**

### CODE REVIEW

**Comment by arpanetus** [comment]
мында қалай скуош жасаса болад 🤔

File: app/commands/eval/service.go [hunk 1]
```diff
@@ -9,7 +9,7 @@ import (
 // CommandCallback is the callback of the "eval" command.
 func CommandCallback(botAPI *tbot.BotAPI, msg tbot.Update) error {
 	if rand.Intn(100) == 0 {
-		text := "вурвур гей"
+		text := "free helicopter rides"
```
Comment (akali) [inline app/commands/eval/service.go:12]: please create another PR for this change

File: app/youtube/model.go [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+)
+
+const (
+	MP4  VideoType = "video/mp4"
+	MKV  VideoType = "video/mkv"
+	WEBM VideoType = "video/webm"
+)
+
+var (
+	ytLinkRegex    = "(((?:https?:)?\\/\\/)?((?:www|m)\\.)?((?:youtube\\.com))(\\/(shorts\\/))([\\w\\-]+)(\\S+)?)"
```
Comment (akali) [inline app/youtube/model.go:37]: What about youtu.be links?

File: app/youtube/model.go [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+}
+
+func (yt *Youtube) Match(text string) []string {
+	return yt.pattern.FindAllString(text, -1)
+}
+
+func (yt *Youtube) chooseFormat(formats youtube.FormatList) *youtube.Format {
+	formats = formats.WithAudioChannels()
+
+	isFailedToFind := true
```
Comment (akali) [inline app/youtube/model.go]: No need for a separate flag, just set chosenFormat as formats[0] in the beginning

Comment (akali) [reply app/youtube/model.go]: Or even directly return format if you able to find, otherwise return &formats[0]

**Review by akali — CHANGES_REQUESTED** [review]

File: app/youtube/model.go (hunk 2 above)
Comment (arpanetus) [reply app/youtube/model.go:37]: why would u need to download another yt videos, it was made for shorts and only

File: app/commands/eval/service.go (hunk 1 above)
Comment (arpanetus) [reply app/commands/eval/service.go:12]: this is canon

**Comment by akali** [comment]
Exactly for shorts sometimes it is been shared with youtu.be link

File: app/youtube/model.go (hunk 2 above)
Comment (arpanetus) [reply app/youtube/model.go:37]: no one shares yt shorts like youtu.be, lets be honest

Comment (akali) [reply app/youtube/model.go:37]: Whatever

**Comment by arpanetus** [comment]
@akali қабылдаш енді

**Pull request merged**